__init__.py

```diff
@@ -66,76 +100,68 @@ def special_day(self):
         else:
             self.speak_dialog("maybe")
 
-    @intent_file_handler("Laugh.intent")
-    def handle_laugh_intent(self, message):
+    @intent_handler("Laugh.intent")
+    def handle_laugh_intent(self, message: Message) -> None:  # noqa
+        """Handle the laugh intent."""
         self.laugh()
 
-    @intent_file_handler("RandomLaugh.intent")
-    def handle_random_intent(self, message):
-        # initiate random laughing
+    @intent_handler("RandomLaugh.intent")
+    def handle_random_intent(self, message: Message) -> None:  # noqa
+        """Initiate random laughing."""
         self.log.info("Laughing skill: Triggering random laughing")
         self.random_laugh = True
         self.handle_laugh_event(message)
 
-    @intent_handler(
-        IntentBuilder('StopLaughing').require('Stop').require('Laugh'))
-    def halt_laughing(self, message):
+    @intent_handler(IntentBuilder("StopLaughing").require("Stop").require("Laugh"))
+    def halt_laughing(self, message: Message) -> None:
+        """Stop the random laughing."""
         self.log.info("Laughing skill: Stopping")
         # if in random laugh mode, cancel the scheduled event
         if self.random_laugh and not self.special_day():
             self.log.info("Laughing skill: Stopping random laugh event")
             self.random_laugh = False
-            self.cancel_scheduled_event('random_laugh')
+            self.cancel_scheduled_event("random_laugh")
             self.speak_dialog("cancel")
             # if haunted == True it will be back on reboot ;)
         else:
             self.speak_dialog("cancel_fail")
 
-    def handle_laugh_event(self, message):
-        # create a scheduled event to laugh at a random interval between 1
-        # minute and half an hour
+    def handle_laugh_event(self, message: Optional[Message]) -> None:
+        """Create a scheduled event for random laughing."""
         if not self.random_laugh:
             return
         self.log.info("Laughing skill: Handling laugh event")
         self.laugh()
-        self.cancel_scheduled_event('random_laugh')
-        self.schedule_event(self.handle_laugh_event,
-                            datetime.now() + timedelta(
-                                seconds=random.randrange(200, 10800)),
-                            name='random_laugh')
-
-    def stop(self, message=None):
-        if self.p is not None:
-            self.p.terminate()
-            return True
-        return False
-
-
-def create_skill():
-    return LaughSkill()
+        self.cancel_scheduled_event("random_laugh")
+        self.schedule_event(
+            self.handle_laugh_event,
+            datetime.now() + timedelta(seconds=random.randrange(200, 10800)),
+            name="random_laugh",
+        )
+
+    def stop(self, message: Message) -> Literal[True]:  # noqa
+        """Stop the audio."""
+        self.send_stop_signal("mycroft.audio.service.stop")
+        return True
```

Comment:
should only return True if we consumed the stop signal, this can potentially hijack the stop mechanism in weird ways


eg, if this skill is the top active skill, no other skills will be stopped since this always consumes "the stop utterance" until this skill is no longer active

Reply:
actually stop method is not needed at all, as the speech service will consume the stop signal itself  (sounds from self.play_audio are treated like TTS)

this bus message would also be wrong since audio service is not used